specs/kyber/src/field.rs

```diff
@@ -0,0 +1,75 @@
+use crate::parameters;
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub(crate) struct FieldElement {
+    pub value : u16
+}
+
+impl FieldElement {
+    const MODULUS : u16 = parameters::FIELD_MODULUS;
+    pub const ZERO: Self = Self { value: 0 };
+
+    pub fn from_u16(inp : u16) -> Self {
+        Self { value : inp % Self::MODULUS }
+    }
+
+    pub fn from_u32(inp : u32) -> Self {
+        Self { value : (inp % u32::from(Self::MODULUS)).try_into().unwrap() }
+    }
+
+    pub fn add(&self, other : &Self) -> Self {
+        let sum : u32 = u32::from(self.value) + u32::from(other.value);
+        Self::from_u32(sum)
+    }
+
+    pub fn subtract(&self, other : &Self) -> Self {
```

Comment:
To make the spec easier to read you can implement the operations and use them in the spec.